src/Core/Silk.NET.BuildTools/Cpp/Clang.cs

```diff
@@ -202,14 +202,35 @@ static void WriteDiagnosticsThrowOnError(TranslationUnit unit)
             var destInfo = task.ClangOpts.ClassMappings[fileName];
             var indexOfOpenSqBracket = destInfo.IndexOf('[');
             var indexOfCloseSqBracket = destInfo.LastIndexOf(']');
-            var projectName = destInfo.Substring
-                (indexOfOpenSqBracket + 1, indexOfCloseSqBracket - indexOfOpenSqBracket - 1);
-            var className = destInfo.Substring(indexOfCloseSqBracket + 1);
+            var projectNameSplit = destInfo.Substring
+                (indexOfOpenSqBracket + 1, indexOfCloseSqBracket - indexOfOpenSqBracket - 1).Split(':');
+
+            if (projectNameSplit.Length == 0) 
+            {
+                throw new ArgumentException("Missing project name!");
+            }
+
+            if (projectNameSplit.Length > 2)
+            {
+                throw new ArgumentException("Too many splits in the project name!");
+            }
+
+            var projectName = projectNameSplit[0];
+
+            if(projectName != "Core" && coreProfile == null) 
+            {
+                throw new InvalidOperationException("The core profile/source file must come first!");
+            }
+
+            var nativeApiSetName = destInfo.Substring(indexOfCloseSqBracket + 1);
+            var className = projectNameSplit.Length > 1 ? projectNameSplit[1] : coreProfile?.Projects["Core"].Classes[0].ClassName ?? nativeApiSetName;
```

Comment:
(continued from previous comment) It should be illegal for a non-core project to refuse to provide a core class name i.e.
```suggestion
            if (projectName != "Core" && projectNameSplit.Length <= 1)
            {
                throw new InvalidOperationException("The core class name must be provided for extension projects. Example: \"[ProjectName:CoreClassName]ExtensionClassName\"")
            }
            
            var className = projectNameSplit.Length > 1
                ? projectNameSplit[1]
                : nativeApiSetName;
```

AFter this I think `coreProfile` can be entirely removed and by extension all of my concerns in Generator.cs will be addressed